certbot/certbot/_internal/main.py

```diff
@@ -1654,6 +1655,129 @@ def make_or_verify_needed_dirs(config: configuration.NamespaceConfig) -> None:
         util.make_or_verify_dir(hook_dir, strict=config.strict_permissions)
 
 
+def _report_reconfigure_results(renewal_file: str, orig_renewal_conf: configobj.ConfigObj) -> None:
+    """Reports the outcome of certificate renewal reconfiguration to the user.
+
+    :param renewal_file: Path to the cert's renewal file
+    :type renewal_file: str
+
+    :param orig_renewal_conf: Loaded original renewal configuration
+    :type orig_renewal_conf: configobj.ConfigObj
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    try:
+        final_renewal_conf = configobj.ConfigObj(
+            renewal_file, encoding='utf-8', default_encoding='utf-8')
+    except configobj.ConfigObjError:
+        raise errors.CertStorageError(
+            f'error parsing {renewal_file}')
+
+    orig_renewal_params = orig_renewal_conf['renewalparams']
+    orig_renewal_params_set = set(orig_renewal_params.items())
+    final_renewal_params = final_renewal_conf['renewalparams']
+    final_renewal_params_set = set(final_renewal_params.items())
```

Comment:
I have a crash here, after the changes get persisted to disk.

Presumably it has something to do with `--webroot-path` being a `Union[str, Sequence[Any], None]`.

    $ certbot certonly --standalone --http-01-port 8123 -d example.com
    $ certbot reconfigure --cert-name example.com --webroot -w /tmp

    Traceback (most recent call last):
      File "/Users/alex/devel/certbot/venv/bin/certbot", line 33, in <module>
        sys.exit(load_entry_point('certbot', 'console_scripts', 'certbot')())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/alex/devel/certbot/certbot/certbot/main.py", line 19, in main
        return internal_main.main(cli_args)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/alex/devel/certbot/certbot/certbot/_internal/main.py", line 1861, in main
        return config.func(config, plugins)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/alex/devel/certbot/certbot/certbot/_internal/main.py", line 1779, in reconfigure
        _report_reconfigure_results(renewal_file, orig_renewal_conf)
      File "/Users/alex/devel/certbot/certbot/certbot/_internal/main.py", line 1681, in _report_reconfigure_results
        final_renewal_params_set = set(final_renewal_params.items())
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: unhashable type: 'list'  TypeError: unhashable type: 'list'

It would be good to get a test for this, and probably an integration test would be good too!

Reply:
lol yes thanks for catching this in the early review, but yes, I still planned to add tests. Don't know that I would have thought to check this particular one or that I expected it to not Just Work there, so helpful nonetheless!